include/swift/Runtime/AccessibleFunction.h

```diff
@@ -0,0 +1,34 @@
+//===---- AcceesibleFunction.h - Runtime accessible functions ---*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// The runtime interface for functions accessible by name.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_ACCESSIBLE_FUNCTION_H
+#define SWIFT_RUNTIME_ACCESSIBLE_FUNCTION_H
+
+#include "swift/ABI/Metadata.h"
+
+#include <cstdint>
+
+namespace swift {
+namespace runtime {
+
+SWIFT_RUNTIME_STDLIB_SPI const AccessibleFunctionRecord *
+swift_findAccessibleFunction(const char *targetNameStart,
+                             size_t targetNameLength);
```

Comment:
We might use flags to detect new scheme of naming if we switch to something other than mangled names.

Reply:
My worry is about both "sides" having to use the same scheme so they'd have to be compiled using the same flags, but I guess that's just what we'll live with 🤔

Reply:
Yeah, that's the unfortunate reality of the situation.

Reply:
Heh yeah... that's what I kept bringing up as "we change it later" being a problem... so we'll be forever stuck emitting the mangled type names as identifiers, and may be able to–in addition to those–add some shorter names... Unless we know "all sides" of the peers and can decide on a scheme to use with a compilation flag... I guess there's ways out of this corner but it'll be annoying :-/

Reply:
Yeah, I think if we just wanted to drop all of the type information then it would be pretty easy to handle at runtime because we could just read mangled name from the record, remove the type information and match against the name provided by the caller. We'd only have to do that transformation once per method and cache it, but for other schemes it would be tricker.

Reply:
Hmmm, I wonder if we could get away with recording using the "full name" and then compare the mangled name if we have it to disambiguate any potential stores with the same name... 🤔 


Not a blocker for merging this, but just something to continue thinking about L(